Yatzy/Category.cs

```diff
@@ -33,46 +46,61 @@ public int Yatzy()
             int total = isYatzy ? 50 : 0;
             return total;
         }
+        
+        private int GetNumbersPoints(int categoryNumber)
+        {
+            bool isNumberPresent = _diceNumbers.Any(number => number == categoryNumber);
+
+            return isNumberPresent ? _diceNumbersWithCount[categoryNumber] * categoryNumber : 0;
+        }
 
         public int Ones()
         {
-            return Numbers(1);
+            return GetNumbersPoints(1);
         }
 
         public int Twos()
         {
-            return Numbers(2);
+            return GetNumbersPoints(2);
         }
 
         public int Threes()
         {
-            return Numbers(3);
+            return GetNumbersPoints(3);
         }
 
         public int Fours()
         {
-            return Numbers(4);
+            return GetNumbersPoints(4);
         }
 
         public int Fives()
         {
-            return Numbers(5);
+            return GetNumbersPoints(5);
         }
 
         public int Sixes()
         {
-            return Numbers(6);
+            return GetNumbersPoints(6);
         }
-        
+     
         public int Pair()
         {
-            return GetPairPoints(1);
-
+            int totalScore = 0;
+            var possiblePairs = GetNumbersWithMultipleOccurence(2);
+            return possiblePairs.Any() ?  possiblePairs.Max() * 2 : 0;
         }
 
         public int TwoPair()
         {
-            return GetPairPoints(2);
+            int totalScore = 0;
+            var possiblePairs = GetNumbersWithMultipleOccurence(2);
+            if (possiblePairs.Count() == 2)
+            {
+                totalScore = possiblePairs.Aggregate((result, value) => result + value) * 2;
+            }
+
+            return totalScore;
```

Comment:
`return (possiblePairs.Count() == 2) ? (possiblePairs.Aggregate((result, value) => result + value) * 2) : 0;`

again `totalScore` is not required if we do that.